### ISSUE TITLE
Runtime API "show stat typed desc" no longer shows descriptions

### ISSUE THREAD

**Issue by NickMRamirez** [haproxy/haproxy#1291]
## Detailed description of the problem

A feature was [introduced in version 2.1](https://www.haproxy.com/blog/haproxy-2-1/#runtime-api-field-descriptions) that allows you to add `desc` to the Runtime API command `show stat typed`, which displays a description for each metric. A sample of the output:

```
$ echo "show stat typed desc" | sudo socat stdio /var/run/haproxy.sock
...
F.2.0.33.rate.1:MRP:u32:0:"Total number of sessions processed by this object over the last second (sessions for listeners/frontends, requests for backends/servers)"
F.2.0.34.rate_lim.1:CLP:u32:0:"Limit on the number of sessions accepted in a second (frontend only, 'rate-limit sessions' setting)"
F.2.0.35.rate_max.1:MMP:u32:0:"Highest value of 'rate' observed since the worker process started"
F.2.0.39.hrsp_1xx.1:MCP:u64:0:"Total number of HTTP responses with status 100-199 returned by this object since the worker process started"
F.2.0.40.hrsp_2xx.1:MCP:u64:0:"Total number of HTTP responses with status 200-299 returned by this object since the worker process started"
...
```

However, this no longer works. The command now returns no descriptions:

```
$ echo "show stat typed desc" | sudo socat stdio /var/run/haproxy.sock
F.2.0.0.pxname.1:KNS:str:stats:"pxname"   
F.2.0.1.svname.1:KNS:str:FRONTEND:"svname"
F.2.0.4.scur.1:MGP:u32:0:"scur"
F.2.0.5.smax.1:MMP:u32:0:"smax"
F.2.0.6.slim.1:CLP:u32:60000:"slim"       
F.2.0.7.stot.1:MCP:u64:0:"stot"
F.2.0.8.bin.1:MCP:u64:0:"bin"
F.2.0.9.bout.1:MCP:u64:0:"bout"
F.2.0.10.dreq.1:MCP:u64:0:"dreq"
F.2.0.11.dresp.1:MCP:u64:0:"dresp"        
F.2.0.12.ereq.1:MCP:u64:0:"ereq"
F.2.0.17.status.1:SGP:str:OPEN:"status" 
```

The feature seems to work up until version 2.2, but not afterwards.

## Expected behavior

Adding `desc` to the end of the `show stat typed` command should add a description of each field to the output.

## Steps to reproduce the behavior

1. Using a version of HAProxy newer than version 2.2, call the `show stat typed desc` command.
2. Note that it does not display descriptions.

## What is your configuration?

relevant parts here to enable the Runtime API:

```
global
    log /dev/log local0 debug
    stats socket /var/run/haproxy.sock user haproxy group haproxy mode 660 level admin
    master-worker
    maxconn 60000
    strict-limits
```

## Output of `haproxy -vv` and `uname -a`

```
$ haproxy -vv
HAProxy version 2.5-dev0-9abb31-178 2021/06/16 - https://haproxy.org/
Status: development branch - not safe for use in production.
Known bugs: https://github.com/haproxy/haproxy/issues?q=is:issue+is:open
Running on: Linux 4.15.0-144-generic #148-Ubuntu SMP Sat May 8 02:33:43 UTC 2021 x86_64
Build options :
  TARGET  = linux-glibc
  CPU     = generic
  CC      = cc
  CFLAGS  = -O2 -g -Wall -Wextra -Wdeclaration-after-statement -fwrapv -Wno-unused-label -Wno-sign-compare -Wno-unused-parameter -Wno-clobbered -Wno-missing-field-initializers -Wtype-limits -Wshift-negative-value -Wshift-overflow=2 -Wduplicated-cond -Wnull-dereference
  OPTIONS = USE_PCRE=1 USE_OPENSSL=1 USE_LUA=1 USE_ZLIB=1 USE_SYSTEMD=1 USE_PROMEX=1
  DEBUG   =

Feature list : +EPOLL -KQUEUE +NETFILTER +PCRE -PCRE_JIT -PCRE2 -PCRE2_JIT +POLL +THREAD +BACKTRACE -STATIC_PCRE -STATIC_PCRE2 +TPROXY +LINUX_TPROXY +LINUX_SPLICE +LIBCRYPT +CRYPT_H +GETADDRINFO +OPENSSL +LUA +ACCEPT4 -CLOSEFROM +ZLIB -SLZ +CPU_AFFINITY +TFO +NS +DL +RT -DEVICEATLAS -51DEGREES -WURFL +SYSTEMD -OBSOLETE_LINKER +PRCTL +THREAD_DUMP -EVPORTS -OT -QUIC +PROMEX -MEMORY_PROFILING

Default settings :
  bufsize = 16384, maxrewrite = 1024, maxpollevents = 200

Built with multi-threading support (MAX_THREADS=64, default=2).
Built with OpenSSL version : OpenSSL 1.1.1  11 Sep 2018
Running on OpenSSL version : OpenSSL 1.1.1  11 Sep 2018
OpenSSL library supports TLS extensions : yes
OpenSSL library supports SNI : yes
OpenSSL library supports : TLSv1.0 TLSv1.1 TLSv1.2 TLSv1.3
Built with Lua version : Lua 5.3.3
Built with the Prometheus exporter as a service
Built with network namespace support.
Built with zlib version : 1.2.11
Running on zlib version : 1.2.11
Compression algorithms supported : identity("identity"), deflate("deflate"), raw-deflate("deflate"), gzip("gzip")
Built with transparent proxy support using: IP_TRANSPARENT IPV6_TRANSPARENT IP_FREEBIND
Built with PCRE version : 8.39 2016-06-14
Running on PCRE version : 8.39 2016-06-14
PCRE library supports JIT : no (USE_PCRE_JIT not set)
Encrypted password support via crypt(3): yes
Built with gcc compiler version 7.5.0

Available polling systems :
      epoll : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use epoll.

Available multiplexer protocols :
(protocols marked as <default> cannot be specified using 'proto' keyword)
              h2 : mode=HTTP       side=FE|BE     mux=H2       flags=HTX|CLEAN_ABRT|HOL_RISK|NO_UPG
            fcgi : mode=HTTP       side=BE        mux=FCGI     flags=HTX|HOL_RISK|NO_UPG
       <default> : mode=HTTP       side=FE|BE     mux=H1       flags=HTX
              h1 : mode=HTTP       side=FE|BE     mux=H1       flags=HTX|NO_UPG
       <default> : mode=TCP        side=FE|BE     mux=PASS     flags=
            none : mode=TCP        side=FE|BE     mux=PASS     flags=NO_UPG

Available services : prometheus-exporter
Available filters :
        [SPOE] spoe
        [CACHE] cache
        [FCGI] fcgi-app
        [COMP] compression
        [TRACE] trace

$ uname -a
Linux lb1 4.15.0-144-generic #148-Ubuntu SMP Sat May 8 02:33:43 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by wtarreau** [comment]
Hi Nick!  Good catch, indeed it got broken.

I could bisect it to this commit: ee63d4bd6 ("MEDIUM: stats: integrate static proxies stats in new stats")

I suspect that the bogus chunk is this one below:
```
@@ -593,8 +598,12 @@ static int stats_dump_fields_typed(struct buffer *out,
                        return 0;
                if (!stats_emit_typed_data_field(out, &stats[field]))
                        return 0;
-               if ((flags & STAT_SHOW_FDESC) && !chunk_appendf(out, ":\"%s\"", stat_fields[field].desc))
+
+               if (flags & STAT_SHOW_FDESC
+                   && !chunk_appendf(out, ":\"%s\"", stat_f[domain][field].name)) {
                        return 0;
+               }
+
                if (!chunk_strcat(out, "\n"))
                        return 0;
        }

```
And indeed, if I change `.name` back to `.desc` it works again. But given that this hunk does nothing else, I don't understand why it had to be done, and I strongly suspect that it's just a leftover of a previous attempt at doing something else, or a thinko.

OK now fixed with commit fc8e43863. Marking for backport.

**Comment by wtarreau** [comment]
fixed in 2.3 as well, closing now.
